example-ideal.txt

```diff
@@ -0,0 +1,11 @@
+go left until blocked on left then
+go up until blocked on up and left then
+(search) go down until blocked on down and left then
+go right then
+go up until blocked on up then
+go right then
+go to search  
+
+Also legal in the language would be:
+go left until blocked on left and down and open on up then
```

Comment:
We touched a bit on this in our earlier discussion about the use of natural language in code, but to reiterate: do you think having the option to express commands in multiple ways might make it more difficult to understand what's going on? As the article we read earlier in the year described, it seems very difficult in practice to substitute code entirely with natural language. Our group decided to simplify the code by incorporate some elements of natural language, but retaining some novel aspects of code. The approach is definitely interesting (although the external implementation probably is a better representation of what you two were trying to go for).